common4j/src/main/com/microsoft/identity/common/java/authscheme/PopAuthenticationSchemeWithClientKeyInternal.java

```diff
@@ -0,0 +1,144 @@
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.authscheme;
+
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SerializedNames.CLIENT_CLAIMS;
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SerializedNames.HTTP_METHOD;
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SerializedNames.KID;
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SerializedNames.NONCE;
+import static com.microsoft.identity.common.java.authscheme.PopAuthenticationSchemeInternal.SerializedNames.URL;
+
+import com.google.gson.annotations.SerializedName;
+import com.nimbusds.jose.util.Base64URL;
+
+import org.json.JSONObject;
+
+import java.net.URL;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import lombok.EqualsAndHashCode;
+import lombok.NonNull;
+import lombok.experimental.Accessors;
+import lombok.experimental.SuperBuilder;
+
+/**
+ * Internal representation of PoP Authentication Scheme where Key is owned by
+ * Client application.
+ */
+@SuperBuilder(toBuilder = true)
+@EqualsAndHashCode(callSuper = true)
+@Accessors(prefix = "m")
+public class PopAuthenticationSchemeWithClientKeyInternal
+        extends TokenAuthenticationScheme
+        implements IPoPAuthenticationSchemeParams {
+
+    private static final long serialVersionUID = 788393037295696359L;
+    /**
+     * The name of this auth scheme as supplied in the Authorization header value.
+     */
+    public static final String SCHEME_POP_WITH_CLIENT_KEY = "PoP_With_Client_Key";
+
+    @SerializedName(HTTP_METHOD)
+    private String mHttpMethod;
+
+    @SerializedName(URL)
+    private URL mUrl;
+
+    @SerializedName(NONCE)
+    private String mNonce;
+
+    @SerializedName(CLIENT_CLAIMS)
+    private String mClientClaims;
+
+    @SerializedName(KID)
+    private String mKid;
+
+    /**
+     * Constructor for gson use.
+     */
+    PopAuthenticationSchemeWithClientKeyInternal() {
+        super(SCHEME_POP_WITH_CLIENT_KEY);
+    }
+
+    /**
+     * Constructs a new PopAuthenticationSchemeWithClientKeyInternal.
+     *
+     * @param httpMethod   The HTTP method associated with this request. Optional.
+     * @param url          The resource URL of future-recipient of this SHR.
+     * @param nonce        Client nonce value; for replay protection.
+     * @param clientClaims Optional claims provided by the caller to embed in the client_claims
+     *                     property of the resulting SHR.
+     */
+    public PopAuthenticationSchemeWithClientKeyInternal(@Nullable final String httpMethod,
```

Comment:
I saw the `SuperBuilder` annotation but see the constructor as well. Is this constructor needed? Should we just stick to the builder instead?

Reply:
I am fine either way, and also keeping both.
I see in oneauth they are using the constructor today for PopAuthenitcationSchemeInternal.